src/physics/physics.cpp

```diff
@@ -653,16 +654,55 @@ void initPhysics() {
 	trackShape->setSimulationFilterData(trackSimFilterData);
 
 	PxRigidStatic* track = gPhysics->createRigidStatic(trackShape->getLocalPose());
+	track->setName("Track");
 	track->attachShape(*trackShape);
 	gScene->addActor(*track);
 
 	// Create vehicle that will drive on the plane. (This one is the player)
 	initVehicle();
 }
 
+int caster = 0;
+
 void stepPhysics() {
 	const PxF32 timestep = 1.0f / 60.0f;
 
+	if (caster++ == 60) {
+		// do raycast
+
+		PxU32 maxHits = 1;
+		PxHitFlags hitFlags = PxHitFlag::ePOSITION | PxHitFlag::eNORMAL | PxHitFlag::eUV;
+
+		physx::PxVehicleDrive4W* player = CTbikes[0];
+		auto origin = player->getRigidDynamicActor()->getGlobalPose().p;
+		auto quat = player->getRigidDynamicActor()->getGlobalPose().q;
+
+		// You can replpace this with basis vectors I think (i forgot about them)
+		// https://www.gamedev.net/forums/topic/56471-extracting-direction-vectors-from-quaternion/1273785
+		float x = 2 * (quat.x * quat.z + quat.w * quat.y);
```

Comment:
As code comment suggests you should be able to switch x, y, z out with the vehicle basis vectors (I think, if basis vectors get updated) to just use the forward vector